### PR TITLE
Improve `utils.FatalError` error formatting

### PR DESCRIPTION
**Pull request by strideynet**
I've noticed the past few months a number of cases where users reporting an issue with `tbot` were confused by the error message output by `tbot`.

A contributing factor to this is that they appear "scrambled". This scrambling appears to be caused by `utils.formatErrorWriter` which prints the Messages from deepest to shallowest, and then prints the underlying error. But - this makes no sense - the underlying error is the deepest element. You can see an example of this below.

I noticed that when `-d` was provided, a different implementation was used to print the error. This implementation is within our `trace` package, and correctly shows the messages from shallowest to deepest - and also indents each message. This helps distinguish each message and is especially important when printing a stack including an aggregate error. So - I figured it made sense to reuse that implementation.

Example Code:

```go
package main

import (
	"github.com/gravitational/teleport/lib/utils"
	"github.com/gravitational/trace"
)

func chargeCapacitors() error {
	return trace.BadParameter("capacitor bank exploded")
}

func prepareEnergySubsystem() error {
	return trace.Wrap(chargeCapacitors(), "charging capacitors")
}

func jumpToLightspeed() error {
	return trace.Wrap(prepareEnergySubsystem(), "preparing energy subsystem")
}

func main() {
	err := trace.Wrap(jumpToLightspeed(), "jumping to lightspeed")
	utils.FatalError(err)
}

```

Prior to this PR:
```
noah@Noahs-MBP teleport % go run ./example   
ERROR: charging capacitors
preparing energy subsystem
jumping to lightspeed
capacitor exploded

exit status 1
```

After this PR:
```
noah@Noahs-MBP teleport % go run ./example 
ERROR: jumping to lightspeed
        preparing energy subsystem
                charging capacitors
                        capacitor exploded

exit status 1
```

If we feel the risk this change is too wide for `utils.FatalError`, I'm more than happy to isolate this fix to `tbot`.

nb: I'm also currently investigating an issue within `trace` when pretty printing aggregate errors. I hope to raise a PR for that soonish as part of a general effort to improve `tbot`'s error output.

### CODE REVIEW

**Review by tigrato — APPROVED** [review]

File: lib/utils/cli.go [hunk 1]
```diff
@@ -203,22 +197,15 @@ func formatErrorWriter(err error, w io.Writer) {
...
-		for _, message := range traceErr.Messages {
-			fmt.Fprintln(w, AllowWhitespace(message))
-		}
-		fmt.Fprintln(w, AllowWhitespace(trace.Unwrap(traceErr).Error()))
-		return
-	}
-	strErr := err.Error()
+
+	msg := trace.UserMessage(err)
 	// Error can be of type trace.proxyError where error message didn't get captured.
```
Comment (codingllama) [inline lib/utils/cli.go:202]: Is this comment still relevant?

File: lib/utils/cli.go [hunk 2]
```diff
@@ -178,21 +178,15 @@ func UserMessageFromError(err error) string {
 // The error message is escaped if necessary. A newline is added if the error text
 // does not end with a newline.
 func FormatErrorWithNewline(err error) string {
```
Comment (codingllama) [inline lib/utils/cli.go:180]: Should we capture the changes in a simple test? Do you think that's worth it?

**Review by codingllama — APPROVED** [review]
Seems reasonable to me, as is the change in error messages overall.

File: lib/utils/cli.go (hunk 2 above)
Comment (strideynet) [reply lib/utils/cli.go:180]: Fixed an existing test

**Pull request merged**